batch/batch/driver/main.py

```diff
@@ -1014,13 +1015,21 @@ async def check(tx):
     CAST(COALESCE(SUM(state = 'Creating' AND cancelled), 0) AS SIGNED) AS actual_n_cancelled_creating_jobs
   FROM
   (
-    SELECT batches.user, jobs.state, jobs.cores_mcpu, jobs.inst_coll,
-      (jobs.always_run OR NOT (jobs.cancelled OR job_groups_cancelled.id IS NOT NULL)) AS runnable,
-      (NOT jobs.always_run AND (jobs.cancelled OR job_groups_cancelled.id IS NOT NULL)) AS cancelled
-    FROM batches
-    INNER JOIN jobs ON batches.id = jobs.batch_id
-    LEFT JOIN job_groups_cancelled ON batches.id = job_groups_cancelled.id
-    WHERE batches.`state` = 'running'
+    SELECT job_groups.user, jobs.state, jobs.cores_mcpu, jobs.inst_coll,
+      (jobs.always_run OR NOT (jobs.cancelled OR t.cancelled IS NOT NULL)) AS runnable,
+      (NOT jobs.always_run AND (jobs.cancelled OR t.cancelled IS NOT NULL)) AS cancelled
+    FROM job_groups
+    LEFT JOIN jobs ON job_groups.batch_id = jobs.batch_id AND job_groups.job_group_id = jobs.job_group_id
+    LEFT JOIN LATERAL (
+      SELECT 1 AS cancelled
+      FROM job_group_self_and_ancestors
+      INNER JOIN job_groups_cancelled
+        ON job_group_self_and_ancestors.batch_id = job_groups_cancelled.id AND
+           job_group_self_and_ancestors.ancestor_id = job_groups_cancelled.job_group_id
+      WHERE job_groups.batch_id = job_group_self_and_ancestors.batch_id AND
+            job_groups.job_group_id = job_group_self_and_ancestors.job_group_id
+    ) AS t ON TRUE
```

Comment:
Same comment.

Reply:
See https://github.com/hail-is/hail/pull/14282#discussion_r1487618770